Fix staff override score

awaisdar001 opened the pull request:
If a learner has not accessed/attempted the score override functionality
didn't work. This PR is intended to fix this behaviour and the
override should work regardless of learner has accessed or attempted
a problem or not.

efischer19 commented on lms/djangoapps/instructor_task/tasks_helper/module_state.py:
```diff
@@ -406,3 +392,39 @@ def _get_task_id_from_xmodule_args(xmodule_instance_args):
+
+def _get_modules_to_update(course_id, usage_keys, student_identifier, filter_fcn, override_score_task):
+    """
+    Fetches a StudentModule instances for a given `course_id`, `student` object, and `usage_keys`.
```
Nit: document more clearly here which params are optional vs required

efischer19 commented on lms/djangoapps/instructor_task/tasks_helper/module_state.py:
```diff
@@ -406,3 +392,39 @@ def _get_task_id_from_xmodule_args(xmodule_instance_args):
+    """
+    def get_student():
+        """
+        Fetches student instance if an identifier is provided and
```
Nit: `Fetches student instance if an identifier is provided, else return None`

awaisdar001 commented:
@efischer19 I have updated the PR with doc string and added an additional check for `student is not None` that we cannot create `StudentModule` entry with `None`  object.

efischer19 commented on lms/djangoapps/courseware/models.py:
```diff
@@ -149,6 +149,18 @@ def __repr__(self):
+        """
+        Return all model instances that correspond to a course and module keys.
+
+        Student ID is optional keyword argument, if provided it narrow narrows down the instances.
```
Super tiny nit: `narrow narrows`

efischer19 commented on lms/djangoapps/instructor_task/tasks_helper/module_state.py:
```diff
@@ -33,19 +34,12 @@ def perform_module_state_update(update_fcn, filter_fcn, _entry_id, course_id, ta
+    the module pointed to by the module_state_key, the particular StudentModule to update, the
+    xmodule_instance_args, and the task_input being passed through.  If the value returned by the
+    update function evaluates to a boolean True, the update is successful; False indicates the update
+    on the particular student module failed.
```
👍

sandroroux commented on lms/djangoapps/instructor_task/tests/test_tasks.py:
```diff
@@ -334,6 +354,25 @@ def test_overriding_success(self):
 
+    def test_overriding_success_with_no_state(self):
+        """
+        Tests that score override is successful for a learner even he has no state
```
`Tests that score override is successful for a learner when they have no state.`
